app/models/concerns/phone_number_validator.rb

```diff
@@ -6,7 +6,7 @@ def validate_each(record, attribute, value)
     return if value.blank?
 
     phone = Phonelib.parse(value, 'US')
-    return unless phone.national(false).nil? || phone.national(false).length != 10
+    return unless phone.e164.blank? || phone.e164.sub(/^\+1+/, '').length != 10
```

Comment:
This regex seems off - I don't think we want the `+` at the end that allows multiple instances of `+1` to be replaced. As it is now, it considers this valid:
![Screen Shot 2021-02-16 at 11 29 44 AM](https://user-images.githubusercontent.com/11698457/108092106-5cb21e80-704a-11eb-963d-a820ed49d748.png)

Reply:
It looks like in this case this happens because the `e164` value created by `Phonelib` does not contain the extra `+` sign. So the value that is passed to the regex ends up being:
```
+117031234567
```
which ends up being allowed because we ignore the preceding `1`s. We are being somewhat permissive in that we allow phone numbers as long as they can be parsed to a e164 representation that starts with a `1`. So something like:
```
+1 234 567 8900
```
would be allowed since the spaces are ignored. We could tighten up the validation so that we literally only allow numbers of the form `+1yxxxxxxxxx` where all the `x`'s are digits, and `y` is not allowed to be `1`. Then it is up to controllers to make sure numbers come in formatted this way (which both the import controller and API controller do currently do). Thoughts? @tstrass?

Reply:
Ah you're right that is parsed out before it's even hit by the regex. I don't think we need to be more strict in the form we handle since PhoneLib handles so much for us while adding extra flexibility for users. I'd honestly be okay leaving as is since it's filtered out by PhoneLib. I guess I'm just wondering why we have the extra `+` at the end of this regex then?

Reply:
This grew out of this conversation here: https://github.com/SaraAlert/SaraAlert/pull/582#discussion_r567327502. The extra `+` makes it so that we don't incorrectly say that a number like:
```
+1 123 456 7890
```
is valid, since that number should not be valid, since an area code cannot start with `1` in the US.